src/complex_fluids/CFINSForcing.cpp

```diff
@@ -279,56 +281,51 @@ CFINSForcing::setDataOnPatchHierarchy(const int data_idx,
             hier_data_ops_manager->getOperationsDouble(d_W_cc_var, hierarchy, true);
         if (d_adv_diff_integrator->getCurrentCycleNumber() == 0 || !W_new_is_allocated)
         {
-            hier_cc_data_ops->copyData(d_W_cc_idx, W_current_idx);
+            hier_cc_data_ops->copyData(d_W_scratch_idx, W_current_idx);
         }
         else
         {
-            hier_cc_data_ops->linearSum(d_W_cc_idx, 0.5, W_current_idx, 0.5, W_new_idx);
+            hier_cc_data_ops->linearSum(d_W_scratch_idx, 0.5, W_current_idx, 0.5, W_new_idx);
         }
     }
 
+    typedef HierarchyGhostCellInterpolation::InterpolationTransactionComponent InterpolationTransactionComponent;
+    std::vector<InterpolationTransactionComponent> ghost_cell_components(1);
+    ghost_cell_components[0] = InterpolationTransactionComponent(d_W_scratch_idx,
+                                                                 "CONSERVATIVE_LINEAR_REFINE",
+                                                                 true,
+                                                                 "CONSERVATIVE_COARSEN",
+                                                                 "LINEAR",
+                                                                 false,
+                                                                 d_conc_bc_coefs,
+                                                                 NULL,
+                                                                 d_interp_type);
+    HierarchyGhostCellInterpolation ghost_fill_op;
+    ghost_fill_op.initializeOperatorState(ghost_cell_components, hierarchy);
+    ghost_fill_op.fillData(data_time);
+
     HierarchyDataOpsManager<NDIM>* hier_data_ops_manager = HierarchyDataOpsManager<NDIM>::getManager();
     Pointer<HierarchyDataOpsReal<NDIM, double> > hier_cc_data_ops =
         hier_data_ops_manager->getOperationsDouble(d_W_cc_var, hierarchy);
     switch (d_evolve_type)
     {
     case SQUARE_ROOT:
-        squareMatrix(d_W_scratch_idx, d_W_cc_var, hierarchy, data_time, initial_time, coarsest_ln, finest_ln);
+        squareMatrix(d_W_scratch_idx, d_W_cc_var, hierarchy, data_time, initial_time, coarsest_ln, finest_ln, true);
```

Comment:
Could you add a comment in front of the hard-coded boolean values to indicate the parameter name?